### PR TITLE
Add tests for Functions 5.2, update output format and add some cases 

### PR DESCRIPTION
**Pull request by Rende11**

### CODE REVIEW

**Comment by plynchnlm** [comment]
Thanks.  Could you separate the 5.2 tests (at least the failing ones) into a separate pull request?  I am planning to work on the remaining 5.1 functions, and would like to be able to merge this into that branch to provide tests for those (and I am not sure I will have time to work on 5.2 features yet).

File: test/cases/functions.yaml [hunk 1]
```diff
@@ -198,7 +249,29 @@ tests:
...
+  - desc: '** if input coll is empty - result of superset is false'
+    expression: Functions.attrempty.supersetOf(Functions.attrdouble)
+    result: [false]
+
+  - desc: '** if input coll isnt exists  - result of superset is false'
+    expression: Functions.attrempty.supersetOf(Functions.nothing)
+    result: [false]
+
+  - desc: '** if argument coll is superset of input coll result is true'
+    expression: Functions.coll1[0].coll2.attr.subsetOf(Functions.attrdouble)
```
Comment (plynchnlm) [inline test/cases/functions.yaml]: supersetOf

File: test/cases/functions.yaml [hunk 2]
```diff
@@ -198,7 +249,29 @@ tests:
...
+    result: [true]
+
+  - desc: '** if argument coll isnt superset of input coll result is false'
+    expression: Functions.attrdouble.subsetOf(Functions.coll1[0].coll2.attr)
```
Comment (plynchnlm) [inline test/cases/functions.yaml]: supersetOf

File: test/cases/functions.yaml [hunk 3]
```diff
@@ -209,29 +282,200 @@ tests:
...
+  - desc: '** isDistinct of non-exists coll is true'
+    expression: Functions.nothing.isDistinct()
+    result: [true]
+
+  - desc: '** isDistinct of distinct coll'
+    expression: Functions.coll1[0].coll2.attr.isDistinct()
+    result: [true]
+
+  - desc: '** isDistinct of coll with different values'
+    expression: Functions.coll1[2].collwithfalse.attr.isDistinct()
```
Comment (plynchnlm) [inline test/cases/functions.yaml]: These are distinct, aren't they?  I see one false and one true.

**Comment by plynchnlm** [comment]
I think I have all the remaining functions for 5.1 done, so no need to split up those tests.  Thanks!

File: test/cases/functions.yaml (hunk 1 above)
Comment (Rende11) [reply test/cases/functions.yaml]: fixed

File: test/cases/functions.yaml (hunk 2 above)
Comment (Rende11) [reply test/cases/functions.yaml]: fixed

File: test/cases/functions.yaml (hunk 3 above)
Comment (Rende11) [reply test/cases/functions.yaml]: fixed

**Pull request merged**